src/pages/search-page/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { useParams } from 'react-router'
+import { useAppContext } from '../../shared/api/app-context-api/app-context-api'
+import { useSearchNewsQuery } from '../../shared/api/news-api'
+import ArticlesSkeleton from '../../shared/ui/articles-skeleton'
+import TopHeadlines from '../../widgets/top-headlines'
+import s from './config/main.module.css'
+
+const SearchPage = () => {
+  const { country } = useAppContext()
+  const { q } = useParams()
+  const { data, isLoading } = useSearchNewsQuery(
```

Comment:
Можно настраивать данные как можно выше/раньше, чтобы до jsx долетали подготовленные уже.
```suggestion
  const { data = [], isLoading } = useSearchNewsQuery(
```
https://redux-toolkit.js.org/rtk-query/usage/queries#query-loading-state
https://redux-toolkit.js.org/rtk-query/usage/queries#selecting-data-from-a-query-result